.circleci/config.yml

```diff
@@ -1,23 +1,458 @@
+      MATCH_APP_IDENTIFIER: 'com.mattermost.rn.NotificationService,com.mattermost.rn.MattermostShare,com.mattermost.rn'
+      EXTENSION_APP_IDENTIFIER: 'com.mattermost.rn.MattermostShare'
+      NOTIFICATION_SERVICE_IDENTIFIER: 'com.mattermost.rn.NotificationService'
+      IOS_APP_GROUP: 'group.com.mattermost'
+      IOS_ICLOUD_CONTAINER: 'iCloud.com.mattermost.rn'
+      IOS_BUILD_EXPORT_METHOD: 'app-store'
+      AWS_FOLDER_NAME: 'mattermost-mobile'
+      MATTERMOST_WEBHOOK_URL: ${MATTERMOST_WEBHOOK_BUILD}
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost.ipa"
+
+  build-ios-pr:
+    executor: ios
+    environment:
+      APP_NAME: "Mattermost Beta"
+      BUILD_PR: true
+      BRANCH_TO_BUILD: ${CIRCLE_BRANCH}
+      REPLACE_ASSETS: false
+      MAIN_APP_IDENTIFIER: "com.mattermost.rnbeta"
+      SENTRY_ENABLED: false
+      MATCH_TYPE: adhoc
+      MATCH_APP_IDENTIFIER: 'com.mattermost.rnbeta.NotificationService,com.mattermost.rnbeta.MattermostShare,com.mattermost.rnbeta'
+      EXTENSION_APP_IDENTIFIER: 'com.mattermost.rnbeta.MattermostShare'
+      NOTIFICATION_SERVICE_IDENTIFIER: 'com.mattermost.rnbeta.NotificationService'
+      IOS_APP_GROUP: 'group.com.mattermost.rnbeta'
+      IOS_ICLOUD_CONTAINER: 'iCloud.com.mattermost.rnbeta'
+      IOS_BUILD_EXPORT_METHOD: 'ad-hoc'
+      AWS_FOLDER_NAME: 'mattermost-mobile-pr'
+      MATTERMOST_WEBHOOK_URL: ${MATTERMOST_WEBHOOK_PR}
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost_Beta.ipa"
+
+  deploy-android-release:
+    executor:
+      name: android
+      resource_class: medium
+    environment:
+      APP_NAME: "Mattermost"
+      BETA_BUILD: false
+    steps:
+      - deploy-android:
+          apk_path: Mattermost.apk
+
+  deploy-android-beta:
+    executor:
+      name: android
+      resource_class: medium
+    environment:
+      APP_NAME: "Mattermost Beta"
+      BETA_BUILD: true
+    steps:
+      - deploy-android:
+          apk_path: Mattermost_Beta.apk
+
+  deploy-ios-release:
+    executor: ios
+    environment:
+      APP_NAME: "Mattermost"
+      BETA_BUILD: false
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost.ipa
+
+  deploy-ios-beta:
+    executor: ios
+    environment:
+      APP_NAME: "Mattermost Beta"
+      BETA_BUILD: true
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost_Beta.ipa
 
 workflows:
   version: 2
-  pr-test:
+  build:
     jobs:
       - test
+
+      - build-android-release:
+          context: mattermost-mobile
+          requires:
+            - test
+          filters:
+            branches:
+              only: /^build-(?!ios|pr|number).*/
+      - deploy-android-release:
+          context: mattermost-mobile
+          requires:
+            - build-android-release
+          filters:
+            branches:
+              only: /^build-(?!ios|pr|number).*/
+
+      - build-android-beta:
+          context: mattermost-mobile
```

Comment:
I'd have a different context on beta than on production, and maybe even differents for android and ios, that way if somehow one attacker gets access to it, wouldn't have all the info accesible.
also, it allows you to have your commands/steps oblivious to what environment is being produced. for them is all the same variables with different values.